### PR TITLE
jarvis automation test cases

### PR DESCRIPTION
**Pull request by sigadamvenkata**
jarvis automation test cases
Jira: MWPW-122175
test page: https://main--milo--adobecom.hlx.live/drafts/nala/features/jarvis/jarvis?georouting=off

### CODE REVIEW

File: tests/milo/javis.feature.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     console.info(`[Test Page]: ${baseURL}${features[0].path}`);
+      await test.step('Navigate to page ', async () => {
+      await page.goto(`${baseURL}${features[0].path}`);
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[0].path}`);
+    });
+  
+      await test.step('Validate jarvis icon present in page', async () => {
+      // added explicit wait since jarvis take around ~5 seconds to load 
+      await page.waitForTimeout(6000);
```
Comment (JackySun9) [inline tests/milo/javis.feature.test.js:22]: better to use waitForSelector

File: tests/milo/javis.feature.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { expect, test } from '@playwright/test';
+import { features } from '../../features/milo/jarvis.spec.js';
+import Jarvis from '../../selectors/milo/jarvis.feature.page.js';
+
+let jarvischat;
+
+test.describe('jarvis features test suite', () => {
+ test.beforeEach(async ({ page }) => {
+    jarvischat = new Jarvis(page);
```
Comment (JackySun9) [inline tests/milo/javis.feature.test.js:9]: name convention, jarvisChat

File: selectors/milo/jarvis.feature.page.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        this.page = page;
+
+        // Jarvis locators in page
+    
+        this.jarvischaticon = page.locator('//button[@id="adbmsgCta"]');
+        this.jarvischaticonclick = page.getByRole('button' , {id : "adbmsgCta"});
+        this.jarvistooltip = page.locator('//*[@aria-label="Chat with us"]');
+        this.javisstartchat = page.locator('//button[@id="adbmsgContentContainer"]');
+        this.jarvisiframewindow = page.locator('//div[@id="adbmsgContentContainer"]');
+        this.jarvishelptext = page.locator('//*[contains(text(), "How can I help you")]');
```
Comment (JackySun9) [inline selectors/milo/jarvis.feature.page.js:14]: This might need to change when move to non-English, do you have css or xpath

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: tests/milo/javis.feature.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+     test(`${features[0].name}, ${features[0].tags}`, async ({ page, baseURL }) => {
```
Comment (amauch-adobe) [inline tests/milo/javis.feature.test.js:12]: this line is indented one too many.

File: tests/milo/javis.feature.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+     console.info(`[Test Page]: ${baseURL}${features[0].path}`);
```
Comment (amauch-adobe) [inline tests/milo/javis.feature.test.js:13]: this line needs to be indented one more time.

File: tests/milo/javis.feature.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      await page.waitForTimeout(6000);
+      await expect(jarvischat.jarvischaticon).toBeVisible();
+      expect(jarvischat.jarvistooltip).toBeTruthy();
+    });
+
+  });
+
+test(`${features[1].name},${features[1].tags}`, async ({ page, baseURL }) => {
+    console.info(`[Test Page]: ${baseURL}${features[1].path}`);
+    await test.step('Validate the jarvis button launch the chat window', async () => {
```
Comment (amauch-adobe) [inline tests/milo/javis.feature.test.js:31]: small nitpick, the word "launch" should be changed to "launches"

File: tests/milo/javis.feature.test.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    await page.goto(`${baseURL}${features[1].path}`);
+    await page.waitForLoadState('domcontentloaded');
+    await expect(page).toHaveURL(`${baseURL}${features[1].path}`);
+    });
+
+    await test.step('Validate the jarvis button launch the chat window', async () => {
```
Comment (amauch-adobe) [inline tests/milo/javis.feature.test.js:37]: both test steps are the exact same, one should be changed. My guess is it should be the first test step.  Probably the first step should be "'Navigate to page ', async () => {".

File: tests/milo/javis.feature.test.js (hunk 1 above)
Comment (amauch-adobe) [reply tests/milo/javis.feature.test.js:22]: I agree as well.

File: tests/milo/javis.feature.test.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      // added explicit wait since jarvis take around ~5 seconds to load 
+      await page.waitForTimeout(6000);
```
Comment (amauch-adobe) [inline tests/milo/javis.feature.test.js:39]: same comment here from @JackySun9, its better to use waitForSelector.

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
see my small comments.  Great job by the way.

**Comment by chadsunvice** [comment]
Before starting the review, I am curious what value do these checks bring, as we already have all the Jarvis use cases tested [here](https://github.com/adobecom/nala/blob/main/tests/feds/jarvis.desktop.test.js). 

I see this as an overlap in logic. Also, the checks in the FEDS folder cover all the use cases, not just that the component loads. It interacts with it and checks it on Mobile landscape, and in different layouts, with different Jarvis configurations. 

This is the full-suite of use cases we already cover: [Sharepoint Jarvis Component](https://adobe.sharepoint.com/:f:/r/sites/adobecom/Shared%20Documents/milo/libs/feds/drafts/qa/[jarvis](https://adobe.sharepoint.com/:f:/r/sites/adobecom/Shared%20Documents/milo/libs/feds/drafts/qa/jarvis?csf=1&web=1&e=Y7Szwm)?csf=1&web=1&e=Y7Szwm).

I see @amauch-adobe has written the ticket. Why do we need these? Worst case, we can move the Jarvis checks already created under the Milo folder, if that is more akin to our regression strategy. But, I see this as an overlap in logic we already have covered.

**Review by chadsunvice — CHANGES_REQUESTED** [review]
Please see my comment in the main thread.

**Comment by narcis-radu** [comment]
We can reuse the existing tests that @chadsunvice has created and move them to the Milo folder since the feature is part of Milo core. @sigadamvenkata, @amauch-adobe I see the JIRA ticket contains additional use-cases to be handled, like initialising and closing the chat conversation; maybe these tests should belong to chat core functionality, but feel free to enhance the existing chat integrations tests with additional use-cases.

**Comment by amauch-adobe** [comment]
> Before starting the review, I am curious what value do these checks bring, as we already have all the Jarvis use cases tested [here](https://github.com/adobecom/nala/blob/main/tests/feds/jarvis.desktop.test.js).
> 
> I see this as an overlap in logic. Also, the checks in the FEDS folder cover all the use cases, not just that the component loads. It interacts with it and checks it on Mobile landscape, and in different layouts, with different Jarvis configurations.
> 
> This is the full-suite of use cases we already cover: [Sharepoint Jarvis Component](https://adobe.sharepoint.com/:f:/r/sites/adobecom/Shared%20Documents/milo/libs/feds/drafts/qa/%5Bjarvis%5D(https://adobe.sharepoint.com/:f:/r/sites/adobecom/Shared%20Documents/milo/libs/feds/drafts/qa/jarvis?csf=1&web=1&e=Y7Szwm)?csf=1&web=1&e=Y7Szwm).
> 
> I see @amauch-adobe has written the ticket. Why do we need these? Worst case, we can move the Jarvis checks already created under the Milo folder, if that is more akin to our regression strategy. But, I see this as an overlap in logic we already have covered.

@chadsunvice and @narcis-radu when I first created the ticket there wasn't anything yet made for the Jarvis chat.  Thanks for pointing the duplication out, I should have looked back in the PR history, that was my fault for not double-checking.  We certainly don't want to duplicate any work if the group feels like the test cases that were listed in the ticket should be added (and provide value to the testing), I say add them to the previous test(s) if that makes sense instead of creating a new test to maintain.  If they don't bring value, let's not create test cases to just create test cases.  As mentioned above, I created that ticket in the early stages of Jarvis development before functional use cases or test refinement were held.  It was a first crack at what might need to be tested.  For all things speaking the ticket might have just needed to be resolved when the first test(s) for Jarvis was made.

**Comment by narcis-radu** [comment]
@amauch-adobe - no worries, I'm glad this has been cleared out. We'll make some changes in our tests to ensure all Milo core features and blocks are reflected in Nala.

**Comment by sigadamvenkata** [comment]
since same feature AU available in feds nala folder we reuse it. hence closing this PR
@skumar09 @amauch-adobe @chadsunvice @JackySun9

**Pull request closed without merging**